frameworks/MLPlan/exec.py

```diff
@@ -0,0 +1,100 @@
+import logging
+import math
+import os
+import json
+
+from frameworks.shared.callee import call_run, result, output_subdir, utils
+
+log = logging.getLogger(__name__)
+
+def run(dataset, config):
+    log.info("\n**** ML-Plans ****\n")
+
+    is_classification = config.type == 'classification'
+    
+    # Mapping of benchmark metrics to Weka metrics
+    metrics_mapping = dict(
+        acc='ERRORRATE',
+        auc='AUC',
+        logloss='LOGLOSS',
+        f1='F1',
+        r2='R2',
+        rmse='ROOT_MEAN_SQUARED_ERROR',
+        mse='MEAN_SQUARED_ERROR',
+        rmsle='ROOT_MEAN_SQUARED_LOGARITHM_ERROR',
+        mae='MEAN_ABSOLUTE_ERROR'
+    )
+
+    metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if metric is None:
+	    raise ValueError('Performance metric {} is not supported.'.format(config.metric))
+    
+    train_file = dataset.train.path
+    test_file = dataset.test.path
+
+    training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
+    backend = config.framework_params.get('_backend', 'weka')
+
+    if backend == "weka":
+        mem_limit = str(max(config.max_mem_size_mb-1024,2048))
+    else:
+        mem_limit = str(max((config.max_mem_size_mb-1024) / config.cores,2048))
```

Comment:
in `-Xmx{val}M` , val  must be an integer
```suggestion
    if backend == "weka":
        mem_limit = str(max(config.max_mem_size_mb-1024, 2048))
    else:
        mem_limit = str(max(round((config.max_mem_size_mb-1024) / config.cores), 2048))
```